### PR TITLE
Merchant Order Show Page

### PR DESCRIPTION
**Pull request by cjkelling**
(#11 ) Create show page view for merchant orders. Renders only information of the order that pertains to the merchant. Page will not render information regarding other merchants orders.

### CODE REVIEW

File: spec/features/merchant/merchant_orders_spec.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    ItemOrder.create!(order_id: @order.id, item_id: @tire.id, quantity: 1, price: 100, merchant_id: @meg.id)
+    ItemOrder.create!(order_id: @order.id, item_id: @pump.id, quantity: 4, price: 25, merchant_id: @meg.id)
+    ItemOrder.create!(order_id: @order.id, item_id: @paper.id, quantity: 2, price: 20, merchant_id: @mike.id)
+    ItemOrder.create!(order_id: @order.id, item_id: @pencil.id, quantity: 3, price: 2, merchant_id: @mike.id)
+
+    visit merchant_orders_path(@order)
+  end
+
+  it 'I only see the items in the order that are being purchased from my merchant' do
+    expect(current_path).to eq("/merchant/orders/#{@order.id}")
```
Comment (lrs8810) [inline spec/features/merchant/merchant_orders_spec.rb:49]: @cjkelling Do we want to use the path helper you created here?

File: spec/features/merchant/merchant_orders_spec.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    end
+
+    within "#item-#{@pump.id}" do
+      expect(page).to have_content(@pump.name)
+      expect(page).to have_css("img[src*='#{@pump.image}']")
+      expect(page).to have_content(@pump.price)
+      expect(page).to have_content('4')
+    end
+
+    expect(page).to_not have_content(@paper.name)
```
Comment (lrs8810) [inline spec/features/merchant/merchant_orders_spec.rb:71]: Very thorough testing.

File: spec/features/merchant/merchant_dash_spec.rb [hunk 3]
```diff
@@ -2,10 +2,10 @@
 
 require 'rails_helper'
 
-RSpec.describe 'As a logged in Merchant (employee/admin)' do
-  it 'on my dashboard, I see the name and address of the merchant I work for' do
-    meg = Merchant.create(name: "Meg's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80_203)
-    merchant_employee = meg.users.create!(
+describe 'As a logged in Merchant (employee/admin)' do
+  before(:each) do
```
Comment (lrs8810) [inline spec/features/merchant/merchant_dash_spec.rb:6]: Thanks for cleaning up this spec with a before each.

File: app/models/order.rb [hunk 4]
```diff
@@ -16,4 +16,8 @@ def grand_total
   def total_quantity
     item_orders.sum(:quantity)
   end
+
+  def find_order(merchant)
```
Comment (lrs8810) [inline app/models/order.rb:20]: @cjkelling Do we have a test for this yet in the spec/models/order_spec.rb file?

**Review by lrs8810 — APPROVED** [review]
Solid work. I know you mentioned not wanting to pass in two instance variables in the #show action, but I think since it's working at the moment we could add it to the refactor issues list.

File: spec/features/merchant/merchant_dash_spec.rb [hunk 5]
```diff
@@ -2,10 +2,10 @@
```
Comment (freeheeling) [inline spec/features/merchant/merchant_dash_spec.rb:2]: This improves organization and readability. Thanks.

File: spec/features/merchant/merchant_orders_spec.rb [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    visit merchant_orders_path(@order)
+  end
+
+  it 'I only see the items in the order that are being purchased from my merchant' do
+    expect(current_path).to eq("/merchant/orders/#{@order.id}")
+
+    expect(page).to have_content(@user.name.to_s)
+    expect(page).to have_content(@user.address.to_s)
+    expect(page).to have_content(@user.city.to_s)
+    expect(page).to have_content(@user.state.to_s)
```
Comment (freeheeling) [inline spec/features/merchant/merchant_orders_spec.rb:54]: Should we always be converting these to strings when testing?

Comment (cjkelling) [reply spec/features/merchant/merchant_orders_spec.rb:54]: Thats just something rubocop does. I think we can do it either way.

**Review by freeheeling — APPROVED** [review]
Everything appears to be addressed per the user story.

**Pull request merged**